src/khoj/processor/plaintext/plaintext_to_jsonl.py

```diff
@@ -0,0 +1,116 @@
+        input_jsonl_files, input_jsonl_filter, output_file = (
+            self.config.input_files,
+            self.config.input_filter,
+            self.config.compressed_jsonl,
+        )
+
+        # Get Plaintext Input Files to Process
+        all_input_plaintext_files = PlaintextToJsonl.get_jsonl_files(input_jsonl_files, input_jsonl_filter)
+
+        # Extract Entries from specified plaintext files
+        with timer("Parse entries from plaintext files", logger):
+            current_entries = PlaintextToJsonl.convert_plaintext_entries_to_maps(
+                PlaintextToJsonl.extract_plaintext_entries(all_input_plaintext_files)
+            )
+
+        # Split entries by max tokens supported by model
+        with timer("Split entries by max token size supported by model", logger):
+            current_entries = self.split_entries_by_max_tokens(current_entries, max_tokens=256)
+
+        # Identify, mark and merge any new entries with previous entries
+        with timer("Identify new or updated entries", logger):
+            entries_with_ids = TextToJsonl.mark_entries_for_update(
+                current_entries, previous_entries, key="compiled", logger=logger
+            )
+
+        with timer("Write entries to JSONL file", logger):
+            # Process Each Entry from All Notes Files
+            entries = list(map(lambda entry: entry[1], entries_with_ids))
+            plaintext_data = PlaintextToJsonl.convert_entries_to_jsonl(entries)
+
+            # Compress JSONL formatted Data
+            compress_jsonl_data(plaintext_data, output_file)
+
+        return entries_with_ids
+
+    @staticmethod
+    def get_jsonl_files(jsonl_files=None, jsonl_file_filters=None):
+        "Get all jsonl files to process"
+        absolute_jsonl_files, filtered_jsonl_files = set(), set()
+        if jsonl_files:
+            absolute_jsonl_files = {get_absolute_path(jsonl_file) for jsonl_file in jsonl_files}
+        if jsonl_file_filters:
+            filtered_jsonl_files = {
+                filtered_file
+                for jsonl_file_filter in jsonl_file_filters
+                for filtered_file in glob.glob(get_absolute_path(jsonl_file_filter), recursive=True)
+            }
+
+        all_target_files = sorted(absolute_jsonl_files | filtered_jsonl_files)
+
+        files_with_no_plaintext_extensions = {
+            target_files for target_files in all_target_files if not PlaintextToJsonl.is_plaintextfile(target_files)
+        }
+        if any(files_with_no_plaintext_extensions):
+            print(f"[Warning] There maybe non plaintext files in the input set: {files_with_no_plaintext_extensions}")
+
+        logger.debug(f"Processing files: {all_target_files}")
+
+        return all_target_files
+
+    @staticmethod
+    def is_plaintextfile(file: str):
+        "Check if file is plaintext file"
+        return file.endswith(("txt", "md", "rst", "html", "htm", "csv", "tsv", "jsonl", "xml"))
```

Comment:
Can we add `.mbox` to this list as well. This will allow folk syncing their mail offline (via IMAP, POP) to search, chat about their emails as well